cmd/clusterctl/client/cluster/cert_manager_test.go

```diff
@@ -247,6 +248,90 @@ func Test_shouldUpgrade(t *testing.T) {
 			want:            false,
 			wantErr:         false,
 		},
+		{
+			name:          "Version is equal but current version has no build metadata, should upgrade",
+			configVersion: "v1.5.3+h4fd4",
+			args: args{
+				objs: []unstructured.Unstructured{
+					{
+						Object: map[string]interface{}{
+							"metadata": map[string]interface{}{
+								"annotations": map[string]interface{}{
+									clusterctlv1.CertManagerVersionAnnotation: "v1.5.3",
+								},
+							},
+						},
+					},
+				},
+			},
+			wantFromVersion: "v1.5.3",
+			wantToVersion:   "v1.5.3+h4fd4",
+			want:            true,
+			wantErr:         false,
+		},
+		{
+			name:          "Version is equal but different build metadata with hash, should upgrade",
```

Comment:
Can you please also add a test case where the build metadata is non-numeric but equal?
Would be nice to call out that case.